inference/core/workflows/core_steps/supervision_tools/segmentations_slicer.py

```diff
@@ -0,0 +1,398 @@
+    )
+
+    class_agnostic_nms: Union[
+        Optional[bool], WorkflowParameterSelector(kind=[BOOLEAN_KIND])
+    ] = Field(
+        default=False,
+        description="Value to decide if NMS is to be used in class-agnostic mode.",
+        examples=[True, "$inputs.class_agnostic_nms"],
+    )
+    class_filter: Union[
+        Optional[List[str]], WorkflowParameterSelector(kind=[LIST_OF_VALUES_KIND])
+    ] = Field(
+        default=None,
+        description="List of classes to retrieve from predictions (to define subset of those which was used while model training)",
+        examples=[["a", "b", "c"], "$inputs.class_filter"],
+    )
+    confidence: Union[
+        FloatZeroToOne,
+        WorkflowParameterSelector(kind=[FLOAT_ZERO_TO_ONE_KIND]),
+    ] = Field(
+        default=0.4,
+        description="Confidence threshold for predictions",
+        examples=[0.3, "$inputs.confidence_threshold"],
+    )
+    mask_decode_mode: Union[
+        Literal["accurate", "tradeoff", "fast"],
+        WorkflowParameterSelector(kind=[STRING_KIND]),
+    ] = Field(
+        default="accurate",
+        description="Parameter of mask decoding in prediction post-processing.",
+        examples=["accurate", "$inputs.mask_decode_mode"],
+    )
+    tradeoff_factor: Union[
+        FloatZeroToOne,
+        WorkflowParameterSelector(kind=[FLOAT_ZERO_TO_ONE_KIND]),
+    ] = Field(
+        default=0.0,
+        description="Post-processing parameter to dictate tradeoff between fast and accurate",
+        examples=[0.3, "$inputs.tradeoff_factor"],
+    )
+    iou_threshold: Union[
+        FloatZeroToOne,
+        WorkflowParameterSelector(kind=[FLOAT_ZERO_TO_ONE_KIND]),
+    ] = Field(
+        default=0.3,
+        description="Parameter of NMS, to decide on minimum box intersection over union to merge boxes",
+        examples=[0.4, "$inputs.iou_threshold"],
+    )
+
+    slice_width: Union[
+        PositiveInt, WorkflowParameterSelector(kind=[INTEGER_KIND])
+    ] = Field(
+        default=320,
+        description="Width of each slice, in pixels",
+        examples=[320, "$inputs.slice_width"],
+    )
+    slice_height: Union[
+        PositiveInt, WorkflowParameterSelector(kind=[INTEGER_KIND])
+    ] = Field(
+        default=320,
+        description="Height of each slice, in pixels",
+        examples=[320, "$inputs.slice_height"],
+    )
+    overlap_ratio_width: Union[
+        FloatZeroToOne, WorkflowParameterSelector(kind=[FLOAT_ZERO_TO_ONE_KIND])
+    ] = Field(
+        default=0.2,
+        description="Overlap ratio between consecutive slices in the width dimension",
+        examples=[0.2, "$inputs.overlap_ratio_width"],
+    )
+    overlap_ratio_height: Union[
+        FloatZeroToOne, WorkflowParameterSelector(kind=[FLOAT_ZERO_TO_ONE_KIND])
+    ] = Field(
+        default=0.2,
+        description="Overlap ratio between consecutive slices in the height dimension",
+        examples=[0.2, "$inputs.overlap_ratio_height"],
+    )
```

Comment:
Similar suggestion to `detections_slicer` - maybe it would result in better user experience if we calculated slice width/height based on request number of slices.